osv2/index.go

```diff
@@ -0,0 +1,152 @@
+package osv2
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/opensearch-project/opensearch-go/v2"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchapi"
+
+	"github.com/CrowdStrike/opensearchtools"
+)
+
+type IndexRequest struct {
+	Operation opensearchtools.IndexOperation
+	Routing   string
+}
+
+// NewIndexRequest instantiates an empty IndexRequest
+func NewIndexRequest() *IndexRequest {
+	return &IndexRequest{}
+}
+
+// Add an index operation to the IndexRequest.
+func (i *IndexRequest) Add(operation opensearchtools.IndexOperation) *IndexRequest {
+	i.Operation = operation
+	return i
+}
+
+func (i *IndexRequest) WithRouting(routing string) *IndexRequest {
+	i.Routing = routing
+	return i
+}
+
+// Validate the index request based on the types
+func (i *IndexRequest) Validate() error {
```

Comment:
Return `ValidationResults` instead. Same comment as the versionless index request.